File: Basics/While-Loops/Exercise/Resources/cake.py
cake_width = int(input())
cake_length = int(input())
cake_area = cake_length * cake_width
is_stop = False

while cake_area > 0:
    current_cake_pieces = input()

    if current_cake_pieces == "STOP":
        is_stop = True
        break

    current_cake_pieces = int(current_cake_pieces)
    cake_area -= current_cake_pieces

if is_stop and cake_area > 0:
    print(f"{cake_area} pieces are left.")
else:
    print(f"No more cake left! You need {abs(cake_area)} pieces more.")

File: JS-Basics/0-5-While-Loops/Exercises/cake.js
function birthdayCake(input) {
    let cakeArea = Number(input.shift()) * Number(input.shift());
    let i = 0; let eatenPieces = 0;
    while (input[i] != "STOP") {
        let n = Number(input[i]);
        cakeArea -= n; i++;
        if (cakeArea < 0) {
            return `No more cake left! You need ${Math.abs(cakeArea)} pieces more.`;
        }    
    }
    return `${cakeArea} pieces are left.`;
}


console.log(birthdayCake(["10",
"10",
"20",
"20",
"20",
"20",
"21"]));
console.log(birthdayCake(["10",
"2",
"2",
"4",
"6",
"STOP"]));